org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -60,6 +82,102 @@ public void start(BundleContext bundleContext) {
 		initializeJDTOptions();
 		projectsManager = new ProjectsManager(preferenceManager);
 		logInfo(getClass()+" is started");
+		configureProxy();
+	}
+
+	private void configureProxy() {
+		String httpHost = System.getProperty(HTTP_PROXY_HOST);
+		String httpPort = System.getProperty(HTTP_PROXY_PORT);
+		String httpUser = System.getProperty(HTTP_PROXY_USER);
+		String httpPassword = System.getProperty(HTTP_PROXY_PASSWORD);
+		String httpsHost = System.getProperty(HTTPS_PROXY_HOST);
+		String httpsPort = System.getProperty(HTTPS_PROXY_PORT);
+		String httpsUser = System.getProperty(HTTPS_PROXY_USER);
+		String httpsPassword = System.getProperty(HTTPS_PROXY_PASSWORD);
+		String httpsNonProxyHosts = System.getProperty(HTTPS_NON_PROXY_HOSTS);
+		String httpNonProxyHosts = System.getProperty(HTTP_NON_PROXY_HOSTS);
+		if (httpUser != null || httpsUser != null) {
+			try {
+				Platform.getBundle("org.eclipse.core.net").start(Bundle.START_TRANSIENT);
+			} catch (BundleException e) {
+				logException(e.getMessage(), e);
+			}
+			if (httpUser != null && httpPassword != null && !httpUser.isEmpty() && !httpPassword.isEmpty()) {
+				Authenticator.setDefault(new Authenticator() {
+					@Override
+					public PasswordAuthentication getPasswordAuthentication() {
+						return new PasswordAuthentication(httpUser, httpPassword.toCharArray());
+					}
+				});
+			}
+			IProxyService proxyService = getProxyService();
+			if (proxyService != null) {
+				ProxySelector.setActiveProvider(MANUAL);
+				IProxyData[] proxies = proxyService.getProxyData();
+				for (IProxyData proxy:proxies) {
+					if ("HTTP".equals(proxy.getType())) {
+						proxy.setHost(httpHost);
+						proxy.setPort(httpPort == null ? 3128 : Integer.valueOf(httpPort));
+						proxy.setPassword(httpPassword);
+						proxy.setUserid(httpUser);
+					}
+					if ("HTTPS".equals(proxy.getType())) {
+						proxy.setHost(httpsHost);
+						proxy.setPort(httpsPort == null ? 3128 : Integer.valueOf(httpsPort));
+						proxy.setPassword(httpsPassword);
+						proxy.setUserid(httpsUser);
+					}
+				}
+				try {
```

Comment:
Please add a comment why you're re-putting the http.* params, for future maintenance